### PR TITLE
add RC kill switch

### PR DESCRIPTION
**Pull request by LorenzMeier**
@kd0aij Final version of the kill switch.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
@kd0aij This should fix the kill switch operation for you, both in terms of functionality as well as in terms of execution speed.

**Comment by kd0aij** [comment]
Thanks, I wasn't at all familiar with the comm's beween FMU and IO

File: src/modules/sensors/sensor_params.c [hunk 1]
```diff
@@ -2243,6 +2252,25 @@ PARAM_DEFINE_FLOAT(RC_ACRO_TH, 0.5f);
  */
 PARAM_DEFINE_FLOAT(RC_OFFB_TH, 0.5f);
 
+
+/**
+ * Threshold for selecting offboard mode
```
Comment (AndreasAntener) [inline src/modules/sensors/sensor_params.c]: comment lies

**Comment by LorenzMeier** [comment]
@kd0aij A switch always has three states: On, Off (some have center as well) and unconfigured. Your logic put everything other than on into the else case, forcing lockdown off (e.g. also in HIL).

**Comment by kd0aij** [comment]
Thanks Lorenz. After reading manual_control_setpoint.msg, I see that a switch actually has a fourth possible state called SWITCH_POS_MIDDLE, at least in this context.

**Comment by kd0aij** [comment]
Question on implementation: not being a mixed C/C++ guru (and vainly hoping it isn't necessary) I don't understand why it's necessary to have things like switch states defined by macros instead of enums which an IDE recognizes as C symbols. Must be something to do with auto-generated header files which don't reside in the Firmware/src tree.

**Comment by LorenzMeier** [comment]
Because they are generated off an IDL which also supports Python and Java as backend. Our messaging architecture also scales to ROS. See the multiplatform apps for an example.

**Comment by kd0aij** [comment]
Flight tested on FMUv2 quadcopter in manual and ALTCTL modes.
kill switch operates as expected: stops the motors promptly when activated, and motors start back up promptly when deactivated.

**Comment by LorenzMeier** [comment]
Rebased and applied, thanks!

**Comment by LorenzMeier** [comment]
added docs here: http://px4.io/docs/flight-mode-configuration/

**Pull request closed without merging**